optuna/visualization/_optimization_history.py

```diff
@@ -48,35 +58,66 @@ def _get_optimization_history_info_list(
 
     info_list: list[_OptimizationHistoryInfo] = []
     for study in studies:
-        trials = study.get_trials(states=(TrialState.COMPLETE,))
+        trials = study.get_trials()
         label_name = target_name if len(studies) == 1 else f"{target_name} of {study.study_name}"
+        values = []
+        value_states = []
+        for trial in trials:
+            if trial.state != TrialState.COMPLETE:
+                values.append(float("nan"))
+                value_states.append(_ValueState.Incomplete)
+                continue
+            has_constraints = _CONSTRAINTS_KEY in trial.system_attrs
+            constraints_value = trial.system_attrs[_CONSTRAINTS_KEY] if has_constraints else [0]
+            value_states.append(
+                _ValueState.Feasible
+                if all(map(lambda x: x <= 0.0, constraints_value))
+                else _ValueState.Infeasible
+            )
```

Comment:
Can we simplify this logic like below?

```suggestion
            constraints = trial.system_attrs.get(_CONSTRAINTS_KEY)
            if constraints is None or all([x <= 0.0 x for x in constraints]):
                value_states.append(_ValueState.Feasible)
            elsel:
                value_states.append(_ValueState.Infeasible)
```

Reply:
This one is simpler and better. I will fix it.